<root BTCPP_format="4">
    <TreeNodesModel>
        <Action ID="Alignment">
            <input_port name="set_play" type="std::string"/>
        </Action>
        <Action ID="ApplySkill"/>
        <Condition ID="BallFound"/>
        <Condition ID="BallFree"/>
        <Condition ID="BallLost"/>
        <Condition ID="CanSaveShot"/>
        <Action ID="Engage"/>
        <Action ID="FootAlignment"/>
        <Action ID="GameStatusProvider">
            <output_port name="set_play" type="std::string"/>
            <output_port name="is_kicking_team" type="bool"/>
            <output_port name="game_state" type="std::string"/>
            <output_port name="game_phase" type="std::string"/>
        </Action>
        <Action ID="GetTarget">
            <output_port name="target" type="nomadz_core::Pose2D"/>
            <input_port name="target_type" type="std::string"/>
        </Action>
        <Condition ID="HasBall"/>
        <Condition ID="HasBallLock"/>
        <Condition ID="IsAligned"/>
        <Condition ID="IsHighestActivePlayer"/>
        <Condition ID="IsInsideArea"/>
        <Action ID="Kick"/>
        <Condition ID="Localized"/>
        <Action ID="LogUndefinedState">
            <input_port name="requested_state" type="std::string"/>
            <input_port name="state_name" type="std::string"/>
        </Action>
        <Action ID="LookAtBall"/>
        <Action ID="LookAtDirection">
            <input_port name="look_target_direction" type="float"/>
        </Action>
        <Condition ID="Lost"/>
        <Condition ID="Penalized"/>
        <Action ID="Penalty"/>
        <Action ID="Recovery"/>
        <Action ID="Relax"/>
        <Action ID="RoleProvider">
            <output_port name="player_role" type="std::string"/>
        </Action>
        <Action ID="SaveShot"/>
        <Action ID="SpecialAction">
            <input_port name="special_action_type" type="nomadz_motion_control_msgs::SpecialActionType"/>
        </Action>
        <Condition ID="Standing"/>
        <Action ID="WalkAtSpeed">
            <input_port name="target_speed" type="nomadz_core::Twist2D"/>
        </Action>
        <Action ID="WalkToTarget">
            <input_port name="target" type="nomadz_core::Pose2D"/>
        </Action>
    </TreeNodesModel>
</root>
